### PR TITLE
LinuxKMS: Add initial support for software rendering with Skia

### PR DESCRIPTION
**Pull request by tronical**
This is the initial implementation based on DRM dumb buffers, which cleans up various things at the same time:

 - DRM output is separate from GBM and DRM dumb buffer displays. The latter two re-use the former.
 - The timer refresh for the Vulkan KHR Display rendering lives now with the KHR display code, in hope to be replaced with a better mechanism in the future.

What's missing still is support for partial updates as well as rendering with our own software renderer.

### CODE REVIEW

**Comment by tronical** [comment]
Some Saturday afternoon fun hacking, built on some initial work I started back in December. Support for our software renderer will be a separate PR (https://github.com/slint-ui/slint/tree/simon/linuxkms-swr).

File: internal/backends/linuxkms/display/gbmdisplay.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+pub struct GbmDisplay {
+    pub drm_output: DrmOutput,
+    gbm_surface: gbm::Surface<OwnedFramebufferHandle>,
+    gbm_device: gbm::Device<SharedFd>,
+}
+
+impl GbmDisplay {
+    pub fn new(drm_output: DrmOutput) -> Result<GbmDisplay, PlatformError> {
+        //eprintln!("mode {}/{}", width, height);
```
Comment (ogoffart) [inline internal/backends/linuxkms/display/gbmdisplay.rs:29]: there are several of these comments.  I don't mind them too much if it helps future debugging.

File: internal/backends/linuxkms/renderer/femtovg.rs [hunk 2]
```diff
@@ -30,22 +31,17 @@ struct GlContextWrapper {
...
-                egl_display.size.height
-            )
+        let height: std::num::NonZeroU32 = height.try_into().map_err(|_| {
+            format!("Attempting to create window surface with an invalid height: {}", height)
         })?;
 
-        let display_handle = egl_display.display_handle().unwrap();
-        let window_handle = egl_display.window_handle().unwrap();
+        let display_handle = gbm_display.display_handle().unwrap();
+        let window_handle = gbm_display.window_handle().unwrap();
```
Comment (ogoffart) [inline internal/backends/linuxkms/renderer/femtovg.rs:44]: can this really never panic?

**Review by ogoffart — APPROVED** [review]

File: internal/backends/linuxkms/renderer/femtovg.rs (hunk 2 above)
Comment (tronical) [reply internal/backends/linuxkms/renderer/femtovg.rs:44]: Heh yes, for once the unwrap is safe, as we always return Ok:

https://github.com/slint-ui/slint/blob/b64b58981d6a481ab2bd73f75b234783fff40a01/internal/backends/linuxkms/display/gbmdisplay.rs#L96

**Pull request merged**